### PR TITLE
Upgrade to Unicode 10

### PR DESCRIPTION
**Pull request by josephfrazier**
Closes https://github.com/slevithan/xregexp/issues/215

EDIT: Kudos to @mathiasbynens for doing the real work in https://github.com/slevithan/xregexp/pull/220 and https://github.com/mathiasbynens/node-unicode-data

### CODE REVIEW

File: tools/output/blocks.js [hunk 1]
```diff
@@ -1090,5 +1090,10 @@ module.exports = [
     {
         'name': 'InYijing_Hexagram_Symbols',
         'bmp': '\u4DC0-\u4DFF'
+    },
+    {
+        'name': 'Inundefined',
```
Comment (mathiasbynens) [inline tools/output/blocks.js]: This seems like a bug

Comment (josephfrazier) [reply tools/output/blocks.js]: Good catch! It looks like `unicode-10.0.0` has an "undefined" block, so I'll open an issue in node-unicode-data.

**Comment by josephfrazier** [comment]
Should we release v4.1.0 now that Unicode 10 is supported?

**Comment by slevithan** [comment]
Yes, sounds good.

**Pull request merged**